internal/notifier/azure_helpers.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2025 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notifier
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/fluxcd/notification-controller/api/v1beta3"
+	pkgauth "github.com/fluxcd/pkg/auth"
+	"github.com/fluxcd/pkg/auth/azure"
+	"github.com/fluxcd/pkg/cache"
+	pkgcache "github.com/fluxcd/pkg/cache"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// newManagedIdentityToken is used to attempt credential-free authentication.
+func newManagedIdentityToken(ctx context.Context, proxy, serviceAccountName, providerName, providerNamespace, scope string, tokenClient client.Client, tokenCache *pkgcache.TokenCache) (string, error) {
+	opts := []pkgauth.Option{pkgauth.WithScopes(scope)}
+	if proxy != "" {
+		proxyURL, err := url.Parse(proxy)
+		if err != nil {
+			return "", fmt.Errorf("error parsing proxy URL : %w", err)
+		}
+		opts = append(opts, pkgauth.WithProxyURL(*proxyURL))
+	}
+
+	if serviceAccountName != "" {
+		serviceAccount := types.NamespacedName{
+			Name:      serviceAccountName,
+			Namespace: providerNamespace,
+		}
+		opts = append(opts, pkgauth.WithServiceAccount(serviceAccount, tokenClient))
+	}
+
+	if tokenCache != nil {
+		involvedObject := cache.InvolvedObject{
+			Kind:      v1beta3.ProviderKind,
+			Name:      providerName,
+			Namespace: providerNamespace,
+			Operation: OperationPost,
+		}
+		opts = append(opts, pkgauth.WithCache(*tokenCache, involvedObject))
+	}
+
+	token, err := pkgauth.GetAccessToken(ctx, azure.Provider{}, opts...)
+	if err != nil {
+		return "", fmt.Errorf("failed to get token : %w", err)
+	}
+
+	return token.(*azure.Token).AccessToken.Token, nil
```

Comment:
```suggestion
	return token.(*azure.Token).Token, nil
```

Can be simplified since `*azure.Token` embeds `azcore.AccessToken`